### DISCUSSION TITLE
2.4.6 Stable Release

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#2391] [General]
Way to many changes to mention but a huge thanks to @int2001 @phl0 @AndreasK79 for all their hard work and support of the project.

Highly recommend reading the commit history

## What's Changed
* Fix C+P mistake by @phl0 in https://github.com/magicbug/Cloudlog/pull/2291
* Dev by @magicbug in https://github.com/magicbug/Cloudlog/pull/2292
* [ADIF] Removed IF as it was stopping any importing by @magicbug in https://github.com/magicbug/Cloudlog/pull/2296
* [Gridmap] Added button to clear lines and markers on map by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2298
* Fixes #2295 // Fill station_callsign when provided empty by adif by @int2001 in https://github.com/magicbug/Cloudlog/pull/2297
* [PHP8.2] Updated CI files to work with PHP8.2 by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2300
* Set Mode to SSB when inverted SSB transponder, otherwise to LSB/USB by @int2001 in https://github.com/magicbug/Cloudlog/pull/2293
* [Distances Worked] When clicking bars, QSOs are displayed in a popup by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2303
* BugFix: Initializes $errors before using it by @int2001 in https://github.com/magicbug/Cloudlog/pull/2307
* Dxcluster lookup by @int2001 in https://github.com/magicbug/Cloudlog/pull/2301
* Display Transmit Power in capital letter (W), added space by @rtoenniges in https://github.com/magicbug/Cloudlog/pull/2309
* Strong check for station_id when importing by @int2001 in https://github.com/magicbug/Cloudlog/pull/2312
* Adjusted caption of contest-session-reset by @int2001 in https://github.com/magicbug/Cloudlog/pull/2314
* Cut off an s in case we have only one QSO by @phl0 in https://github.com/magicbug/Cloudlog/pull/2319
* Reset LoTW upload for LEDSAT by @phl0 in https://github.com/magicbug/Cloudlog/pull/2322
* DXCluster integration by @int2001 in https://github.com/magicbug/Cloudlog/pull/2317
* Removed old API-Endpoints for country_ and grid_worked by @int2001 in https://github.com/magicbug/Cloudlog/pull/2323
* Fixed Submode/Mode issue when checking if worked by @int2001 in https://github.com/magicbug/Cloudlog/pull/2324
* Fixed handling of empty continent of spotter by @int2001 in https://github.com/magicbug/Cloudlog/pull/2325
* Bandmap allbands by @int2001 in https://github.com/magicbug/Cloudlog/pull/2327
* since kHZ is no longer part of the table, substring isn't needed by @int2001 in https://github.com/magicbug/Cloudlog/pull/2328
* Added 401 in case of missing key to api/radio for easier debugging by @int2001 in https://github.com/magicbug/Cloudlog/pull/2329
* [Bandmap] Switched over to local copies of Highcharts js files by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2330
* A station is worked, when one OR MORE Times worked... by @int2001 in https://github.com/magicbug/Cloudlog/pull/2332
* Use LoTW as official abbreviation by @phl0 in https://github.com/magicbug/Cloudlog/pull/2333
* Bandmap2qso by @int2001 in https://github.com/magicbug/Cloudlog/pull/2331
* Label fix ascii by @int2001 in https://github.com/magicbug/Cloudlog/pull/2335
* SatMode label fix by @int2001 in https://github.com/magicbug/Cloudlog/pull/2336
* Shortened Date /  by @int2001 in https://github.com/magicbug/Cloudlog/pull/2337
* Russian translation updated by @cats-shadow in https://github.com/magicbug/Cloudlog/pull/2338
* try to fix some things... by @int2001 in https://github.com/magicbug/Cloudlog/pull/2340
* Add code to calculate 4, 6, 8 char grid squares by @magicbug in https://github.com/magicbug/Cloudlog/pull/2339
* Split SAT mode in (adv) logbook and QSO view by @phl0 in https://github.com/magicbug/Cloudlog/pull/2344
* Look also for QSOs where QSL column is NULL or empty by @phl0 in https://github.com/magicbug/Cloudlog/pull/2347
* [Labels] Added a dialog for the user to specify from which label the print should start. by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2349
* Security fixes by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2350
* Added Menu for maintenance by @int2001 in https://github.com/magicbug/Cloudlog/pull/2353
* [Advanced Logbook] Added dialog to set label offset when printing labels by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2354
* protect DANGEROUS Functions from being called without authed user by @int2001 in https://github.com/magicbug/Cloudlog/pull/2355
* CodeIgniter fix to deal with null-values by @int2001 in https://github.com/magicbug/Cloudlog/pull/2356
* Correction of typo [DE] by @205er in https://github.com/magicbug/Cloudlog/pull/2357
* integrate Winkey by @magicbug in https://github.com/magicbug/Cloudlog/pull/2358
* When inserting qso via API (or logging) set get_qso to trusted by @int2001 in https://github.com/magicbug/Cloudlog/pull/2359
* Implemented Userlanguages with a lot of help from @AndreasK79 by @int2001 in https://github.com/magicbug/Cloudlog/pull/2361
* Custom paper types for labels by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2367
* Fixed prefix on SQL by @int2001 in https://github.com/magicbug/Cloudlog/pull/2368
* Added inches to papertype / extended logic for inches by @int2001 in https://github.com/magicbug/Cloudlog/pull/2369
* [QSLPrint] Fixed js. Added button to mark as sent by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2371
* Finnish update by @oh1jw in https://github.com/magicbug/Cloudlog/pull/2370
* [Winkeyer] Fixed url when Cloudlog is in a subfolder by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2372
* Added (optional) grid to label-printing by @int2001 in https://github.com/magicbug/Cloudlog/pull/2375
* Delete HW-Radio worked only for admins. Fixed that by @int2001 in https://github.com/magicbug/Cloudlog/pull/2374
* Think it should be here :) by @phl0 in https://github.com/magicbug/Cloudlog/pull/2379
* Contest Multi-Op (with same Account and Call) by @int2001 in https://github.com/magicbug/Cloudlog/pull/2378
* Updated some Fin and Swe files by @oh1jw in https://github.com/magicbug/Cloudlog/pull/2380
* [QSO Entry] Cleanup and tighten up db lookup from QSO entry by @AndreasK79 in https://github.com/magicbug/Cloudlog/pull/2382
* Added "Already imported to clublog" to ADIF-Importer by @int2001 in https://github.com/magicbug/Cloudlog/pull/2383
* Mark QRZ for reupload if LotW-Confirmation was received after qrz.com by @int2001 in https://github.com/magicbug/Cloudlog/pull/2386
* Same logic for Clublog (ReUp after LoTW-Conf) by @int2001 in https://github.com/magicbug/Cloudlog/pull/2387
* Prevent QSO being updated twice or more when LoTW-Cnf arrives by @int2001 in https://github.com/magicbug/Cloudlog/pull/2389
* Update NULLs at LoTW, too by @int2001 in https://github.com/magicbug/Cloudlog/pull/2390
* Some minor grammatical changes for the German translation by @205er in https://github.com/magicbug/Cloudlog/pull/2366
* PR to create stable release 2.4.6 by @magicbug in https://github.com/magicbug/Cloudlog/pull/2388

## New Contributors
* @rtoenniges made their first contribution in https://github.com/magicbug/Cloudlog/pull/2309
* @205er made their first contribution in https://github.com/magicbug/Cloudlog/pull/2357

**Full Changelog**: https://github.com/magicbug/Cloudlog/compare/2.4.5...2.4.6

<hr /><em>This discussion was created from the release <a href='https://github.com/magicbug/Cloudlog/releases/tag/2.4.6'>2.4.6 Stable Release</a>.</em>

**Comment by MikleBas** [comment]
Hello!
Thanks for your work, I updated without problems.
although this release is marked as stable, but I didn’t have any problems with the previous ones, everything works like a clock :)
PS WinKey didn't work for me, it just puts the transceiver in transmit mode and that's it....

**Reply by magicbug** [reply]
Winkey seems to be having issues with initialisation working on that still it functions better once the Units been started by another application 

If anyone is experienced with webserial I could use some pointers

**Reply by MikleBas** [reply]
When access to USB CAT is not occupied by another program, it just clicks on PTT
but if I run CwType which I use, an error appears (see screenshot)
Maybe this will help you catch the error.
![1](https://github.com/magicbug/Cloudlog/assets/99086477/43f74d66-9499-4e8f-9944-06df9466a076)

**Comment by MikleBas** [comment]
Hello!
All the same, prompts began to work unstable when entering a callsign in the communication window
and the function that showed earlier that connections have already been made is not very stable
At first I thought that the problem was on my part, but then I came to the conclusion that the software was not working stably.
What to pay attention to?
Thank you

**Reply by AndreasK79** [reply]
Are you using HamQTH for callbook? If that is the case, you should switch to the dev branch, since we have a few bugs in the latest release.

**Reply by MikleBas** [reply]
![image](https://github.com/magicbug/Cloudlog/assets/99086477/5690abb3-47ce-4588-b71e-22606949b4ac)

**Reply by magicbug** [reply]
Something strange seems to be going on with lookup as after you enter a call sign the dxcc worked isn't always functional now. Not had time to dig into the why yet

**Reply by AndreasK79** [reply]
Rewriting working stuff is not always a good idea, even though we try to make the software better and more secure. Sorry about the bugs introduced. I think I traced down the problem now. I pushed a fix to dev. Please pull latest dev and try again.
